cluster-autoscaler/config/autoscaling_options.go

```diff
@@ -46,6 +46,52 @@ type NodeGroupAutoscalingOptions struct {
 	ScaleDownUnreadyTime time.Duration
 }
 
+const (
+	// MaxAllocatableDifferenceRatio describes how Node.Status.Allocatable can differ between
+	// groups in the same NodeGroupSet
+	MaxAllocatableDifferenceRatio = 0.05
+	// MaxFreeDifferenceRatio describes how free resources (allocatable - daemon and system pods)
+	// can differ between groups in the same NodeGroupSet
+	MaxFreeDifferenceRatio = 0.05
+	// MaxCapacityMemoryDifferenceRatio describes how Node.Status.Capacity.Memory can differ between
+	// groups in the same NodeGroupSet
+	MaxCapacityMemoryDifferenceRatio = 0.015
+)
+
+// NodeGroupDifferenceRatios contains various ratios used to determine if two NodeGroups are similar and makes scaling decisions
+type NodeGroupDifferenceRatios struct {
+	// MaxAllocatableDifferenceRatio describes how Node.Status.Allocatable can differ between groups in the same NodeGroupSet
+	MaxAllocatableDifferenceRatio float64
+	// MaxFreeDifferenceRatio describes how free resources (allocatable - daemon and system pods) can differ between groups in the same NodeGroupSet
+	MaxFreeDifferenceRatio float64
+	// MaxCapacityMemoryDifferenceRatio describes how Node.Status.Capacity.Memory can differ between groups in the same NodeGroupSetAutoscalingOptions
+	MaxCapacityMemoryDifferenceRatio float64
+}
+
+// GetMaxAllocatableDifferenceRatio returns MaxAllocatableDifferenceRatio if it is set, otherwise returns default value
```

Comment:
Why do the defaulting both here and in flag definitions? This may be confusing, setting any flag explicitly to 0 will not work as expected. I'd suggest dropping the getter functions completely and using the consts above to populate flag defaults. For tests, creating struct populated with defaults would just require a test helper then. WDYT?

Reply:
I agree we should only have a single default. I guess I initially thought maybe someone could leverage these types and not have access to the defaults set in the flags, but looking at how its used now I think your plan makes much more sense because of the zero case.